src/pools/TradePool.cairo

```diff
@@ -0,0 +1,133 @@
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.starknet.common.syscalls import get_caller_address, get_contract_address
+from starkware.cairo.common.bool import FALSE, TRUE
+from starkware.cairo.common.uint256 import (
+    Uint256, 
+    uint256_le, 
+    uint256_sub,
+    uint256_add
+)
+
+from lib.cairo_contracts.src.openzeppelin.token.erc20.IERC20 import IERC20
+
+from src.pools.IPool import NFT
+from src.pools.Pool import PriceUpdate
+from src.pools.Pool import (
+    _current_price,
+    _pool_paused,
+    _eth_balance,
+    _erc20_address,
+)
+from src.pools.Pool import (
+    constructor,
+    setPoolParams,
+    addSupportedCollections,
+    removeSupportedCollections,
+    addNftToPool,
+    _add_nft_to_pool,
+    removeNftFromPool,
+    _remove_nft_from_pool,
+    getAllCollections,
+    getAllNftsOfCollection,
+    get_total_price,
+    get_next_price,
+    assert_collections_supported,
+    togglePause,
+    isPaused,
+    getPoolFactory,
+    getPoolConfig,
+    getNextPrice,
+    getTokenPrices,
+    depositEth,
+    withdrawEth,
+    withdrawAllEth,
+    getStartIdByCollection,
+    getListElementById,
+    getCollectionById,
+    getEthBalance,
+    checkCollectionSupport,
+    assert_only_owner,
+    assert_not_owner
+)
+
+
+@external
+func sellNfts{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    nft_array_len: felt, nft_array: NFT*
+) -> () {
+    alloc_locals;
+    assert_not_owner();
+
+    let (is_paused) = _pool_paused.read();
+    with_attr error_message("Pool must not be paused") {
+        assert is_paused = FALSE;
+    }
+
+    assert_collections_supported(nft_array_len, nft_array);
+
+    let (total_price) = get_total_price(nft_array_len, -1);
+    let (new_price) = get_next_price(nft_array_len, -1);
+
+    let (eth_balance) = _eth_balance.read();
+    let (sufficient_balance) = uint256_le(total_price, eth_balance);
+    with_attr error_message("Pool ETH balance is not sufficient") {
+        assert sufficient_balance = TRUE;
+    }
```

Comment:
Probably, that's a dedicated function as well, if you do the same assert at different places.